cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -733,18 +732,19 @@ void reader::impl::load_and_decompress_data(std::vector<row_group_info> const& r
         total_decompressed_size += col_meta.total_uncompressed_size;
       }
     }
-    // Read compressed chunk data to device memory
-    read_rowgroup_tasks.push_back(read_column_chunks_async(_sources,
-                                                           raw_page_data,
-                                                           chunks,
-                                                           io_chunk_idx,
-                                                           chunks.size(),
-                                                           column_chunk_offsets,
-                                                           chunk_source_map,
-                                                           _stream));
-
     remaining_rows -= row_group.num_rows;
   }
+
+  // Read compressed chunk data to device memory
+  read_rowgroup_tasks.push_back(read_column_chunks_async(_sources,
+                                                         raw_page_data,
+                                                         chunks,
+                                                         0,
```

Comment:
So previously it read `io_chunk_idx` which is const set to `chunks.size()`. Now it is `0`?

Reply:
The `chunks.size()` was captured before adding to chunks for the row group, so the starting size of the chunks array. It would be 0, then however many chunks were there at the next iteration, etc.

The function parameters now say start at chunk 0 and read all the chunks.